Configure external sources

tomusdrw opened the pull request:
Closes: #144 

- [x] Removes export of legacy notes (most likely not used anyway, and if needed they stay in local storage).
- [x] Introduce notes settings dialog allowing to configure:
- [x] - changing activity of remote notes source
- [x] - adding/altering custom remote sources
- [x] Remote notes may only be active in some versions instead of all of them. 
- [x] Splitted element notes into multiple files to allow more fine-grained control over them 
- [x] changed katex delimiter to fix some rendering issues (i.e. incorrect boundaries of formulas).

skoszuta commented on src/components/PdfViewer/PdfViewer.tsx:
```diff
@@ -80,6 +80,7 @@ export function PdfViewer() {
 
       eventBus.on("pagesloaded", () => {
         setPagesLoaded(true);
+        pdfViewer.currentScaleValue = "0.85";
```
pls make a const for this one too

tomusdrw replied:
✅ done

skoszuta reviewed (changes requested):
Code looks good ✅ But there are some issues in the app:
- The remove button not doing anything when adding a new remote: https://www.loom.com/share/f09e66691797425982588d3f280defdd
- There's no button spacing between ok and remove
- Modal content is invisible in dark mode: https://www.loom.com/share/e89a81a533054f0bad3fc4e4b77a7f4f
- `white-space: pre-wrap` is causing some alignment issues in labels: https://www.loom.com/share/f75678c906a74afdb2d61530cd2d7333

tomusdrw commented on src/components/NoteRenderer/components/HighlightNote/HighlightNote.tsx:
```diff
@@ -32,8 +33,21 @@ export function HighlightNote({ notes, pageOffset, isInViewport, isPinnedByDefau
+      // we delay only hover off
+      setHoveredTimeout.current = setTimeout(() => {
+        setHovered(false);
+      }, 350);
```
✅ done

tomusdrw commented:
Thanks for the review, all issues should be addressed now. 
- remove button isn't visible now
- spacing between `ok` and `remove` added.
- fixed dark mode issues and cleaned up colors a bit
- removed excess whitespace from labels to avoid the pre-wrap issues.

skoszuta reviewed (commented):
It looks like the code formatting commit reintroduced the label spacing issue.

tomusdrw commented:
Good catch :) I've added explicit `white-space` to the label span now.

skoszuta reviewed (approved):
lookin good 🚢
